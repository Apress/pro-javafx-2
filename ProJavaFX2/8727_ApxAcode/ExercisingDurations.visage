/*
 * @test
 * @run
 */

var oneHour = 1h;
var oneMinute = 1m;
var oneSecond = 1s;
var oneMillisecond = 1ms;

println("oneHour = {oneHour}");
println("oneMinute = {oneMinute}");
println("oneSecond = {oneSecond}");
println("oneMillisecond = {oneMillisecond}");

println("oneHour + 30 * oneMinute = {oneHour + 30 * oneMinute}");
println("oneMinute - 20 * oneSecond = {oneMinute - 20 * oneSecond}");
println("oneSecond / 2 = {oneSecond / 2}");
println("-oneMillisecond = {-oneMillisecond}");

println("oneHour.toHours() = {oneHour.toHours()}");
println("oneHour.toMinutes() = {oneHour.toMinutes()}");
println("oneMinute.toSeconds() = {oneMinute.toSeconds()}");
println("oneSecond.toMillis() = {oneSecond.toMillis()}");

var now = Duration.valueOf(1229923309734.0);
println("now = {now}");
