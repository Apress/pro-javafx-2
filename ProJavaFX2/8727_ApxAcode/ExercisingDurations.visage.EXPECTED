oneHour = 3600000ms
oneMinute = 60000ms
oneSecond = 1000ms
oneMillisecond = 1ms
oneHour + 30 * oneMinute = 5400000ms
oneMinute - 20 * oneSecond = 40000ms
oneSecond / 2 = 500ms
-oneMillisecond = -1ms
oneHour.toHours() = 1.0
oneHour.toMinutes() = 60.0
oneMinute.toSeconds() = 60.0
oneSecond.toMillis() = 1000.0
now = 1229923287040ms
